my_enumerable.rb

```diff
@@ -0,0 +1,91 @@
+module Enumerable
+  def my_each
+    for item in self do
+      yield(item)
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    for item in self do
+      yield(item, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_arr = []
+    for item in self do
+      new_arr.push(item) if yield(item)
+    end
+    new_arr
+  end
+
+  def my_all?
+    result = true
+    for item in self do
+      return false if yield(item) == false
+    end
+    result
+  end
+
+  def my_any?
+    for item in self do
+      return true if yield(item)
+    end
+    false
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil
```ruby
true_array = [nil, false, true, []]
true_array.my_any? == true_array.any? #true
```
- [ ]  when a class is passed as an argument returns true if at least one of the collection is a member of such class
```ruby
words = %w[dog door rod blade]
words.my_any?(Integer) == words.any?(Integer) #true
```
- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex
```ruby
words.my_any?(/z/) == words.any?(/z/) #true
```
- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```ruby
words.my_any?('cat') == words.any?('cat') #true
```